File: backend/src/main/java/com/ronbodnar/reciperepository/model/RecipeIngredient.java
package com.ronbodnar.reciperepository.model;

import com.ronbodnar.reciperepository.enums.MeasurementType;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
public class RecipeIngredient {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private int id;

    @ManyToOne
    @JoinColumn(name = "recipe_id")
    private Recipe recipe;

    @ManyToOne
    @JoinColumn(name = "ingredient_id")
    private Ingredient ingredient;

    @Column(name = "quantity")
    private double quantity;

    @Column(name = "measurement_type")
    private MeasurementType measurementType;

}

File: frontend/src/app/core/models/recipe-ingredient.ts
import { Ingredient } from "./ingredient";
import { MeasurementType } from "../enums/measurement-type";
import { Recipe } from "./recipe";

export class RecipeIngredient {

    id!: number;

    recipe!: Recipe;

    ingredient!: Ingredient;

    quantity!: number;

    measurementType!: MeasurementType;
}
